### PR TITLE
add section Build complex Data Types in Book

### PR DESCRIPTION
**Pull request by badermarouan**
This section explains what is operator overload and how to use it. It also provides a description of Templated Structs for custom Types and how to use them.

### CODE REVIEW

File: _docs/operators_overload.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+## Using the Overload in Code
+
+Once the operator is overloaded, you can compare the names and ages of two instances of Person as follows:
+
+```glu
+func main() {
+    let Emil : Person = {"Emil", 20};
+    let Jerem : Person = {"Jerem", 20};
+    let EmilClone : Person = {"Emil", 20};
```
Comment (Snowy1803) [inline _docs/operators_overload.md:52]: avoid variable names starting with an uppercase letter

Comment (Snowy1803) [reply _docs/operators_overload.md:52]: also, im 21 not 20 🙄

File: _docs/operators_overload.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        std::print("Emil is not Jerem! ");
+    }
+    if (Emil == EmilClone) {
+        std::print("It's a Clone!!!");
+    }
+}
+```
+In this example the following output should be:
+```
+Emil is not Jerem! It's a Clone!!!
```
Comment (Snowy1803) [inline _docs/operators_overload.md:66]: I think the output should have a newline (std::print should add a newline)

File: _docs/templating_structures.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+---
+
+Templates allow you to define generic structures that can handle different data types without rewriting the structure for each type. This makes your code more reusable and flexible. In our language, the syntax for defining a template structure is straightforward. Here’s how to use template structures.
+
+# Defining a Template Structure
+
+Let's define a template structure Container that can hold any type of data:
+
+```glu
+struct<T> Container {
```
Comment (Snowy1803) [inline _docs/templating_structures.md:13]:
```suggestion
struct Container<T> {
```

File: _docs/templating_structures.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+We can create a template structure to hold a Person instance:
+
+```glu
+let person : Person = { "Alice", 30 };
+let personContainer : Container<Person> = { person };
+std::print(personContainer.content.name); // Output: Alice
+std::print(personContainer.content.age);  // Output: 30
+```
+
+Conclusion:
```
Comment (Snowy1803) [inline _docs/templating_structures.md:76]:
```suggestion
## Conclusion
```

File: _docs/templating_structures.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
+---
+ title: Templating Structures
+ category: Templates
+---
+
+Templates allow you to define generic structures that can handle different data types without rewriting the structure for each type. This makes your code more reusable and flexible. In our language, the syntax for defining a template structure is straightforward. Here’s how to use template structures.
+
+# Defining a Template Structure
```
Comment (Snowy1803) [inline _docs/templating_structures.md:8]:
```suggestion
## Defining a Template Structure
```

File: _docs/templating_structures.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+```glu
+struct<T> Container {
+    content: T
+}
+```
+
+Here's an explanation of the components:
+
+## 1. Template Declaration:
```
Comment (Snowy1803) [inline _docs/templating_structures.md:20]:
```suggestion
### Template Declaration:
```
etc.

File: _docs/operators_overload.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
+---
+ title: Operators overload
```
Comment (Snowy1803) [inline _docs/operators_overload.md:2]:
```suggestion
 title: Overloading Operators
```
starting with a verb is what we do in other articles

File: _docs/operators_overload.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+ category: Data Types
+---
+
+Operator overloading allows you to redefine the behavior of operators for specific data types. This makes your code more intuitive and readable. In our language, the syntax for overloading an operator is straightforward.
+
+# Example of Overloading the == Operator
```
Comment (Snowy1803) [inline _docs/operators_overload.md:8]: you could add examples for other operators, and add a link to the appendix with a list of all operators

File: _docs/operators_overload.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    let emilClone : Person = {"Emil", 21};
+
+    if (Emil == Jerem) {
+        std::print("Emil is Jerem??");
+    } else {
+        std::print("Emil is not Jerem! ");
+    }
+    if (Emil == EmilClone) {
+        std::print("It's a Clone!!!");
+    }
```
Comment (Snowy1803) [inline _docs/operators_overload.md:61]: update the names of the variables

File: _docs/operators_overload.md (hunk 8 above)
Comment (badermarouan) [reply _docs/operators_overload.md:8]: added in the conclusion

**Review by Snowy1803 — APPROVED** [review]

**Review by Snowy1803 — APPROVED** [review]

File: _config.yml [hunk 10]
```diff
@@ -83,13 +83,11 @@ collections:
...
-      - comments.md
-      - control_flow.md
-      - structs.md
-      - enums.md
-      - typealiases.md
+      - declare_constants_or_variables.md
+      - comments_in_glu.md
+      - structs_in_glu.md
+      - operators_overload.md
+      - templating_structures.md
```
Comment (Snowy1803) [inline _config.yml:90]: that's no good

**Review by Snowy1803 — CHANGES_REQUESTED** [review]

File: _config.yml (hunk 10 above)
Comment (Snowy1803) [reply _config.yml:90]: the conflicts weren't resolved correctly, those articles were renamed and some are missing.

**Review by Snowy1803 — APPROVED** [review]

**Pull request merged**